src/ComServerHelpers/Internal/BaseActivationFactoryWrapper.cs

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.Marshalling;
+using ComServerHelpers.Windows.Com;
+using Windows.Win32.Foundation;
+using static Windows.Win32.PInvoke;
+
+namespace ComServerHelpers.Internal;
+
+[GeneratedComClass]
+internal sealed partial class BaseActivationFactoryWrapper(BaseActivationFactory factory, ComWrappers comWrappers) : IActivationFactory
+{
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Design", "CA1031:Do not catch general exception types", Justification = "COM method, must not throw.")]
+    public unsafe HRESULT ActivateInstance(void** instance)
+    {
+        if (instance is null)
+        {
+            return HRESULT.E_INVALIDARG;
+        }
+
+        nint unknown = 0;
+        try
+        {
+            object managedInstance = factory.ActivateInstance();
+            factory.OnInstanceCreated(managedInstance);
+
+            unknown = comWrappers.GetOrCreateComInterfaceForObject(managedInstance, CreateComInterfaceFlags.None);
+            *instance = (void*)unknown;
```

Comment:
This still suffers from the stale `out` problem. The following code will fail badly. The invariant is that out parameters with lifetime rules can't be modified if an error code is being returned.

```cpp
CComPtr<IUnknown> pUnk;
HRESULT hr = ActivateInstance(&pUnk);
if (FAILED(hr))
{
    return hr;
}

// When pUnk goes out of scope, the destructor will release the pointer,
// but it was already released in the callee.